static/js/settings_streams.js

```diff
@@ -19,10 +19,20 @@ function failed_listing_streams(xhr) {
 function populate_streams(streams_data) {
     var streams_table = $("#admin_streams_table").expectOne();
     all_streams = streams_data;
-    streams_table.find("tr.stream_row").remove();
-    _.each(streams_data.streams, function (stream) {
-        streams_table.append(templates.render("admin_streams_list", {stream: stream}));
-    });
+
+    list_render(streams_table, streams_data.streams, {
+        name: "admin_streams_list",
+        modifier: function (item) {
+            return templates.render("admin_streams_list", { stream: item });
+        },
+        filter: {
+            element: streams_table.closest(".settings-section").find(".search"),
+            callback: function (item, value) {
+                return item.name.toLowerCase().match(value);
+            },
+        },
+    }).init();
+
```

Comment:
Test-deployed this to chat.zulip.org, and this one works great, unless you add a filter for a character that doesn't exist.

![image](https://cloud.githubusercontent.com/assets/2746074/25301580/da5a1274-26df-11e7-98ed-62bb89a16f93.png)

Also, it appears the `list_render` library doesn't support non-English characters; if you put in something with a `ó`  in it, and then search for that character, nothing matches.